app/models/agents/local_file_agent.rb

```diff
@@ -0,0 +1,186 @@
+    form_configurable :watch, type: :array, values: %w(true false)
+    form_configurable :path, type: :string
+    form_configurable :append, type: :boolean
+    form_configurable :data, type: :string
+
+    def validate_options
+      if options['mode'].blank? || !['read', 'write'].include?(options['mode'])
+        errors.add(:base, "The 'mode' option is required and must be set to 'read' or 'write'")
+      end
+      if options['watch'].blank? || ![true, false].include?(boolify(options['watch']))
+        errors.add(:base, "The 'watch' option is required and must be set to 'true' or 'false'")
+      end
+      if options['append'].blank? || ![true, false].include?(boolify(options['append']))
+        errors.add(:base, "The 'append' option is required and must be set to 'true' or 'false'")
+      end
+      if options['path'].blank?
+        errors.add(:base, "The 'path' option is required.")
+      end
+    end
+
+    def working?
+      should_run?(false) && ((interpolated['mode'] == 'read' && check_path_existance && checked_without_error?) ||
+                             (interpolated['mode'] == 'write' && received_event_without_error?))
+    end
+
+    def check
+      return if interpolated['mode'] != 'read' || boolify(interpolated['watch']) || !should_run?
+      return unless check_path_existance(true)
+      if File.directory?(interpolated['path'])
+        Dir.glob(File.join(interpolated['path'], '*')).select { |f| File.file?(f) }
+      else
+        [interpolated['path']]
+      end.each do |file|
+        create_event payload: get_file_pointer(file)
+      end
+    end
+
+    def receive(incoming_events)
+      return if interpolated['mode'] != 'write' || !should_run?
+      incoming_events.each do |event|
+        mo = interpolated(event)
+        File.open(mo['path'], boolify(mo['append']) ? 'a' : 'w') do |file|
+          file.write(mo['data'])
+        end
+      end
+    end
+
+    def start_worker?
+      interpolated['mode'] == 'read' && boolify(interpolated['watch']) && should_run? && check_path_existance
+    end
+
+    def check_path_existance(log = true)
+      if !File.exist?(interpolated['path'])
+        error("File or directory '#{interpolated['path']}' does not exist") if log
+        return false
+      end
+      true
+    end
+
+    def get_io(file)
+      File.open(file, 'r')
+    end
+
+    private
+
+    def should_run?(log = true)
+      if self.class.should_run?
+        true
+      else
+        error("Unable to run because insecure agents are not enabled. Set ENABLE_INSECURE_AGENTS to true in the Huginn .env configuration.") if log
+        false
+      end
+    end
+
+    class Worker < LongRunnable::Worker
+      def setup
+        require 'listen'
+        @listener = Listen.to(*listen_options, &method(:callback))
+      end
+
+      def run
+        sleep unless agent.check_path_existance(true)
+
+        @listener.start
+        sleep
```

Comment:
Until the `AgentRunner` is stopped or the `Worker` is killed by the AgentRunner, the sleep is needed to keep the worker thread alive. If we would return from the run method the thread would stop.